okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -0,0 +1,831 @@
+   * a network interceptor to short-circuit or repeat a network request.
+   */
+  @get:JvmName("networkInterceptors")
+  val networkInterceptors: List<Interceptor> = Util.immutableList(builder.networkInterceptors)
+
+  @get:JvmName("eventListenerFactory")
+  val eventListenerFactory: EventListener.Factory = builder.eventListenerFactory
+
+  @get:JvmName("proxySelector")
+  val proxySelector: ProxySelector = builder.proxySelector
+
+  @get:JvmName("cookieJar")
+  val cookieJar: CookieJar = builder.cookieJar
+
+  @get:JvmName("cache")
+  val cache: Cache? = builder.cache
+
+  @get:JvmName("socketFactory")
+  val socketFactory: SocketFactory = builder.socketFactory
+
+  @get:JvmName("sslSocketFactory")
+  val sslSocketFactory: SSLSocketFactory
+
+  @get:JvmName("hostnameVerifier")
+  val hostnameVerifier: HostnameVerifier = builder.hostnameVerifier
+
+  @get:JvmName("certificatePinner")
+  val certificatePinner: CertificatePinner
+
+  @get:JvmName("proxyAuthenticator")
+  val proxyAuthenticator: Authenticator = builder.proxyAuthenticator
+
+  @get:JvmName("authenticator")
+  val authenticator: Authenticator = builder.authenticator
+
+  @get:JvmName("connectionPool")
+  val connectionPool: ConnectionPool = builder.connectionPool
+
+  @get:JvmName("dns")
+  val dns: Dns = builder.dns
+
+  @get:JvmName("followSslRedirects")
+  val followSslRedirects: Boolean = builder.followSslRedirects
+
+  @get:JvmName("followRedirects")
+  val followRedirects: Boolean = builder.followRedirects
+
+  @get:JvmName("retryOnConnectionFailure")
+  val retryOnConnectionFailure: Boolean = builder.retryOnConnectionFailure
+
+  /**
+   * Default call timeout (in milliseconds). By default there is no timeout for complete calls, but
+   * there is for the connect, write, and read actions within a call.
+   */
+  @get:JvmName("callTimeoutMillis")
+  val callTimeoutMillis: Int = builder.callTimeout
+
+  /** Default connect timeout (in milliseconds). The default is 10 seconds.  */
+  @get:JvmName("connectTimeoutMillis")
+  val connectTimeoutMillis: Int = builder.connectTimeout
+
+  /** Default read timeout (in milliseconds). The default is 10 seconds. */
+  @get:JvmName("readTimeoutMillis")
+  val readTimeoutMillis: Int = builder.readTimeout
+
+  /** Default write timeout (in milliseconds). The default is 10 seconds. */
+  @get:JvmName("writeTimeoutMillis")
+  val writeTimeoutMillis: Int = builder.writeTimeout
+
+  /** Web socket and HTTP/2 ping interval (in milliseconds). By default pings are not sent. */
+  @get:JvmName("pingIntervalMillis")
+  val pingIntervalMillis: Int = builder.pingInterval
+
+  private val internalCache: InternalCache? = builder.internalCache
+  private val certificateChainCleaner: CertificateChainCleaner
+
+  constructor() : this(Builder())
+
+  init {
+    var isTLS = false
+    for (spec in connectionSpecs) {
+      isTLS = isTLS || spec.isTls
+    }
+
+    // TODO - these nullability warnings existed in Java
```

Comment:
Let me know how you want to handle these

![image](https://user-images.githubusercontent.com/1361086/54588942-90f5ae00-49e1-11e9-8f4e-7828ab1c23c7.png)